numba/runtime/nrt.c

```diff
@@ -7,11 +7,25 @@
 #define MIN(a, b) ((a) < (b)) ? (a) : (b)
 #endif
 
+#ifdef IS_FLAG_SET
+#  error "IS_FLAG_SET redefined"
+#else
+#  define IS_FLAG_SET(X, M) ( (X & M) == M )
+#endif
 
-typedef int (*atomic_meminfo_cas_func)(MemInfo * volatile *ptr, MemInfo *cmp,
-                                       MemInfo *repl, MemInfo **oldptr);
+typedef int (*atomic_meminfo_cas_func)(void **ptr, void *cmp,
+                                       void *repl, void **oldptr);
 
 
+enum MEMINFO_FLAGS{
+    MEMINFO_FLAGS_NONE = 0,
+    /*
+     *  If set, the MemInfo and the data buffer are in the same allocated
+     *  region.  Freeing the MemInfo pointer will free the data.
+     */
+    MEMINFO_FLAGS_INLINED = 1
```

Comment:
I remembered why.  ``NRT_MemInfo_destroy()`` needs to know how to "release" the meminfo object.  If I remove the freelists all together, it can be much simpler.